tinycrypt/ecc.c

```diff
@@ -266,15 +272,76 @@ static void muladd(uECC_word_t a, uECC_word_t b, uECC_word_t *r0,
 
 }
 
-/* Computes result = left * right. Result must be 2 * num_words long. */
-static void uECC_vli_mult(uECC_word_t *result, const uECC_word_t *left,
-			  const uECC_word_t *right, wordcount_t num_words)
+/* State for implementing random delays in uECC_vli_mult_rnd().
+ *
+ * The state is initialised by randomizing delays and setting i = 0.
+ * Each call to uECC_vli_mult_rnd() uses one byte of delays and increments i.
+ *
+ * A scalar muliplication uses 14 field multiplications per bit of exponent.
+ */
+typedef struct {
```

Comment:
Yes, an instance of `wait_state_t` is allocated in the next commit in `uECC_Point_mult()`. I wanted to avoid a huge commit by separating the code implementation of randomized long mult from it integration in point mult, but I agree it has the drawback of declaring types which as not used to allocate anything in the next commit, sorry for that.

Good point about renaming the type. Even though it's internal to this C file, in the future other wait states might become visible here as well.

Reply:
Done.